Framework/Algorithms/src/ApplyTransmissionCorrection.cpp

```diff
@@ -27,116 +27,182 @@ using namespace Geometry;
+} // namespace PropertyNames
+} // namespace
+
 void ApplyTransmissionCorrection::init() {
   auto wsValidator = boost::make_shared<CompositeValidator>();
   wsValidator->add<WorkspaceUnitValidator>("Wavelength");
   wsValidator->add<HistogramValidator>();
+  declareProperty(
+      std::make_unique<WorkspaceProperty<>>(PropertyNames::INPUT_WKSP, "",
+                                            Direction::Input, wsValidator),
+      "Workspace to apply the transmission correction to");
   declareProperty(std::make_unique<WorkspaceProperty<>>(
-                      "InputWorkspace", "", Direction::Input, wsValidator),
-                  "Workspace to apply the transmission correction to");
-  declareProperty(std::make_unique<WorkspaceProperty<>>("TransmissionWorkspace",
-                                                        "", Direction::Output,
-                                                        PropertyMode::Optional),
+                      PropertyNames::TRANSMISSION_WKSP, "", Direction::Output,
+                      PropertyMode::Optional),
                   "Workspace containing the transmission values [optional]");
-  declareProperty(std::make_unique<WorkspaceProperty<>>("OutputWorkspace", "",
-                                                        Direction::Output),
+  declareProperty(std::make_unique<WorkspaceProperty<>>(
+                      PropertyNames::OUTPUT_WKSP, "", Direction::Output),
                   "Workspace to store the corrected data in");
 
   // Alternatively, the user can specify a transmission that will ba applied to
   // all wavelength bins
   declareProperty(
-      "TransmissionValue", EMPTY_DBL(),
+      PropertyNames::TRANSMISSION_VALUE, EMPTY_DBL(),
       "Transmission value to apply to all wavelengths. If specified, "
       "TransmissionWorkspace will not be used.");
   auto mustBePositive = boost::make_shared<BoundedValidator<double>>();
   mustBePositive->setLower(0.0);
-  declareProperty("TransmissionError", 0.0, mustBePositive,
+  declareProperty(PropertyNames::TRANSMISSION_ERROR, 0.0, mustBePositive,
                   "The error on the transmission value (default 0.0)");
 
   declareProperty(
       "ThetaDependent", true,
       "If true, a theta-dependent transmission correction will be applied.");
 }
 
+std::map<std::string, std::string>
+ApplyTransmissionCorrection::validateInputs() {
+  std::map<std::string, std::string> result;
+
+  // needed to find out what mode we are working in
+  const double trans_value = getProperty(PropertyNames::TRANSMISSION_VALUE);
+  if (isEmpty(trans_value)) { // require a transmission workspace
+    MatrixWorkspace_const_sptr transWS =
+        getProperty(PropertyNames::TRANSMISSION_WKSP);
+    if (!transWS) {
+      const std::string msg(
+          "Must specify \"TransmissionValue\" or \"TransmissionWorkspace\"");
+      result[PropertyNames::TRANSMISSION_VALUE] = msg;
+      result[PropertyNames::TRANSMISSION_WKSP] = msg;
```

Comment:
make `msg` mutable and use move it here
`result[PropertyNames::TRANSMISSION_WKSP] = std::move(msg)`